frontend/src/index.pug

```diff
@@ -170,32 +170,48 @@ html
               )
                 span {{ getReportIssueEmailAddress() }}
             div.error-message-box__failed-repo--reason(v-else) {{ errorBlock.errorMessage }}
+        .fileTypes(v-if="filterBreakdown")
+          div.checkboxes.mui-form--inline(v-if="Object.keys(contributionBarFileTypeColors).length > 0")
+            label(
+              v-bind:style="{ 'background-color' : '#000000', 'color' : '#ffffff' }"
+            )
+              input.mui-checkbox--fileType(type="checkbox" v-model="checkAllFileTypes")
+              span
+                span.select.medium-font All:&nbsp;
+            template(v-for="fileType in Object.keys(checkBoxFileTypeFontColors)" )
+              label(
+                v-bind:key="fileType"
+                v-bind:style="{ 'background-color' : contributionBarFileTypeColors[fileType]," +
+                "'color' : checkBoxFileTypeFontColors[fileType]}"
+              )
+                input.mui-checkbox--fileType(
+                  type="checkbox"
+                  v-bind:id="fileType"
+                  v-bind:value="fileType"
+                  v-model="checkedFileTypes"
+                )
+                span {{ fileType }}
         #summary-charts
           .summary-charts(v-for="(repo, i) in filteredRepos")
             .summary-charts__title(v-if="filterGroupSelection === 'groupByRepos'")
               .summary-charts__title--index {{ i+1 }}
               .summary-charts__title--repo {{ repo[0].repoName }}
               .summary-charts__title--contribution.mini-font(title="Total contribution of group") [{{ getGroupTotalContribution(repo) }} lines]
             .summary-charts__title(
-                v-if="filterGroupSelection === 'groupByAuthors'",
-                v-bind:class="{ warn: repo[0].name === '-' }"
+              v-if="filterGroupSelection === 'groupByAuthors'",
+              v-bind:class="{ warn: repo[0].name === '-' }"
             )
               .summary-charts__title--index {{ i+1 }}
               .summary-charts__title--repo {{ repo[0].displayName }} ({{ repo[0].name }})
               .summary-charts__title--contribution.mini-font(title="Total contribution of group") [{{ getGroupTotalContribution(repo) }} lines]
             .summary-charts__fileType--breakdown(v-if="filterBreakdown")
-              template(v-if="filterGroupSelection === 'groupByRepos'")
-                .summary-charts__fileType--breakdown__legend(
-                    v-for="fileType in getFileTypes(repo)")
-                  i.fas.fa-circle(
-                      v-bind:style="{ 'color' : contributionBarFileTypeColors[fileType] }"
-                  )
-                  span &nbsp; {{ fileType }} &nbsp;
-              template(v-else)
+              template
                 .summary-charts__fileType--breakdown__legend(
-                    v-for="fileType in Object.keys(contributionBarFileTypeColors)")
+                  v-if="filterGroupSelection !== 'groupByNone'"
```

Comment:
Vue does not recommend putting v-for and v-if together in the same div, check: https://vuejs.org/v2/guide/conditional.html#v-if-with-v-for